src/AgentImplementation.sol

```diff
@@ -232,7 +300,7 @@ contract AgentImplementation is IAgent, ERC721Holder, ERC1155Holder {
         // TODO: implement new msg.value fee charge mechanism
     }
 
-    function _chargeFee(IParam.Fee[] memory fees) internal {
+    function _chargeFee(IParam.Fee[] memory fees, bool chargeFromUser) internal {
         uint256 length = fees.length;
         if (length == 0) return;
```

Comment:
Though this is not modified in this PR, want to be sure if we need `uint256 length = fees.length`? (Seems `fees.length` is already in memory)

Reply:
Will check.

Reply:
https://sandbox.tenderly.co/hihiben/curved-monitor
Seems to be slightly cheaper.